### PR TITLE
[AIRFLOW-3704] Support SSL Protection When Redis is Used as Broker for CeleryExecutor

### PR DESCRIPTION
**Pull request by XD-DENG**
### Jira

  - https://issues.apache.org/jira/browse/AIRFLOW-3704

### Description

From Celery 4.1 (current Airflow is using 4.1.1), `broker_use_ssl` argument starts to support Redis (earlier this argument is only supported when pyamqp is used for broker) (reference: http://docs.celeryproject.org/en/latest/userguide/configuration.html#broker-use-ssl).

But the details are different when user is using pyaqmp or Redis.

The current implementation will pass wrong arguments when user is using Redis as broker and has chosen SSL_ACTIVE = True. Two possibilities:
- if some exception is raised, user will not be able to SSL protection when they are using Redis, even though this is already supported by Celery.
- this may work and skip wrong configuration silently, and user may think they have set up the SSL protection while it's not working.

Neither is acceptable.

This PR has enabled using SSL protection when Redis is used as broker for CeleryExecutor.

### CODE REVIEW

File: airflow/config_templates/default_celery.py [hunk 1]
```diff
@@ -75,6 +85,6 @@ def _broker_supports_visibility_timeout(url):
                            'SSL and/or have all necessary certs and key ({}).'.format(e))
 
 result_backend = DEFAULT_CELERY_CONFIG['result_backend']
-if 'amqp' in result_backend or 'redis' in result_backend or 'rpc' in result_backend:
+if 'amqp://' in result_backend or 'redis://' in result_backend or 'rpc://' in result_backend:
```
Comment (XD-DENG) [inline airflow/config_templates/default_celery.py:88]: This change is to address cases like `mysql://amqp:password@localhost` (i.e., "amqp" or "redis" may happen to be part of my user name).

It's not really relating to this PR subject. But it may not be worthwhile to open another PR.

**Comment by XD-DENG** [comment]
@ashb @kaxil @feng-tao PTAL. A security improvement.

File: airflow/config_templates/default_celery.py [hunk 2]
```diff
@@ -60,10 +60,20 @@ def _broker_supports_visibility_timeout(url):
...
+        if 'amqp://' in broker_url:
+            broker_use_ssl = {'keyfile': configuration.conf.get('celery', 'SSL_KEY'),
+                              'certfile': configuration.conf.get('celery', 'SSL_CERT'),
+                              'ca_certs': configuration.conf.get('celery', 'SSL_CACERT'),
+                              'cert_reqs': ssl.CERT_REQUIRED}
+        elif 'redis://' in broker_url:
+            broker_use_ssl = {'ssl_keyfile': configuration.conf.get('celery', 'SSL_KEY'),
+                              'ssl_certfile': configuration.conf.get('celery', 'SSL_CERT'),
+                              'ssl_ca_certs': configuration.conf.get('celery', 'SSL_CACERT'),
+                              'ssl_cert_reqs': ssl.CERT_REQUIRED}
```
Comment (ashb) [inline airflow/config_templates/default_celery.py:72]: Celery is annoying :(

**Review by ashb — APPROVED** [review]

File: airflow/config_templates/default_celery.py (hunk 2 above)
Comment (XD-DENG) [reply airflow/config_templates/default_celery.py:72]: Yes, I agree. The same items with different keys...

**Comment by ashb** [comment]
Will merge once tests are green (if I don't please ping me again :) )

**Review by feng-tao — APPROVED** [review]
LGTM
